reconcile/utils/saasherder.py

```diff
@@ -334,6 +334,25 @@ def _additional_resource_process(resources, html_url):
                         'could not add html_url annotation to' +
                         resource['name'])
 
+    @staticmethod
+    def _parameter_required_and_absent(
+            parameter_name, consolidated_parameters, template):
+        """Is a parameter is required in the template but unspecified?
+
+        :param parameter_name: The name (key) of the parameter.
+        :param consolidated_parameters: Dict of parameters already specified/
+                calculated.
+        :param template: The template file in dict form.
+        :return bool: True if the named parameter is required by the template,
+                but not already present in consolidated_parameters.
+        """
+        if parameter_name in consolidated_parameters:
+            return False
+        for template_parameter in template.get("parameters", {}):
+            if template_parameter["name"] == parameter_name:
+                return template_parameter.get("required", False)
```

Comment:
This is new logic. We can't rely on the IMAGE_TAG being required, but we still want to generate it. Can we keep this to a separate PR?

Reply:
Reverted the logic and left a TODO.